suite-native/module-accounts/src/components/AccountRenameForm.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+import { Box, Text, Button, VStack } from '@suite-native/atoms';
+import { TextInputField, Form, useForm } from '@suite-native/forms';
+import { yup } from '@trezor/validation';
+import {
+    accountsActions,
+    AccountsRootState,
+    selectAccountByKey,
+    selectAccountLabel,
+} from '@suite-common/wallet-core';
+
+type AccountRenameFormProps = {
+    accountKey: string;
+    onSubmit: () => void;
+};
+
+const MAX_LABEL_LENGTH = 30;
+
+const accountRenameValidationSchema = yup.object({
+    accountLabel: yup.string().required().max(MAX_LABEL_LENGTH),
+});
+type AccountImportFormValues = yup.InferType<typeof accountRenameValidationSchema>;
+
+export const AccountRenameForm = ({ accountKey, onSubmit }: AccountRenameFormProps) => {
+    const dispatch = useDispatch();
+    const account = useSelector((state: AccountsRootState) =>
+        selectAccountByKey(state, accountKey),
+    );
+    const accountName = useSelector((state: AccountsRootState) =>
+        selectAccountLabel(state, accountKey),
+    );
+
+    const renameForm = useForm<AccountImportFormValues>({
+        validation: accountRenameValidationSchema,
+        defaultValues: {
+            accountLabel: accountName,
+        },
+        mode: 'onChange',
+    });
+    const {
+        handleSubmit,
+        formState: { isValid },
+    } = renameForm;
+
+    if (!account) return null;
+
+    const handleRenameAccount = handleSubmit(({ accountLabel }: AccountImportFormValues) => {
+        dispatch(accountsActions.renameAccount(account, accountLabel));
+        onSubmit();
+    });
+
+    return (
+        <Box marginTop="medium">
+            <Form form={renameForm}>
+                <VStack spacing="small">
+                    <TextInputField name="accountLabel" label="Account label" />
+                    <Box paddingLeft="small">
+                        <Text variant="label" color="gray600">
+                            {renameForm.watch('accountLabel').length} / {MAX_LABEL_LENGTH} letters
```

Comment:
This API is bad because according to docs https://react-hook-form.com/api/useform/watch/
```
This API will trigger re-render at the root of your app or form, consider using a callback or the useWatch api if you are experiencing performance issues.
```

Plase make this text as standalone small component, that will use `useWatch` instead so it won't rerender this whole form on every type of letter (please verify that after you will be done with changes). You can keep component in this file.

Reply:
fixed in e44e7fee9